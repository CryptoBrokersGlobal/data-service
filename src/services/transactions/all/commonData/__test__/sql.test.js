const sql = require('../sql');
const filters = {
  sender: 'sender',
  timeStart: 'timeStart',
  timeEnd: 'timeEnd',
  sort: 'sort',
  limit: 1,
};

describe('Sql builder', () => {
  describe('search', () => {
    it('covers case with all filters (without after)', () => {
      expect(sql.search(filters)).toMatchSnapshot();
    });
    it('covers case with all filters with after', () => {
      expect(
        sql.search({
          ...filters,
          after: {
            timestamp: 'timestamp',
            id: 'id',
            sortDirection: 'sortDirection',
          },
        })
      ).toMatchSnapshot();
    });
  });
  describe('get', () => {
    it('works', () => {
      expect(sql.get('id')).toMatchSnapshot();
    });
  });
  describe('mget', () => {
    it('works', () => {
      expect(sql.mget(['id1', 'id2'])).toMatchSnapshot();
    });
  });
});
