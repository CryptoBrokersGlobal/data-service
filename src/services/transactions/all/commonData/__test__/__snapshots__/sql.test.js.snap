// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sql builder get works 1`] = `"select \\"t\\".\\"tx_type\\" as \\"tx_type\\", \\"t\\".\\"id\\" as \\"id\\" from \\"txs\\" as \\"t\\" where \\"id\\" = 'id'"`;

exports[`Sql builder mget works 1`] = `"select \\"t\\".\\"tx_type\\" as \\"tx_type\\", \\"t\\".\\"id\\" as \\"id\\" from \\"txs\\" as \\"t\\" where \\"id\\" in ('id1', 'id2')"`;

exports[`Sql builder search covers case with all filters (without after) 1`] = `"select \\"t\\".\\"tx_type\\" as \\"tx_type\\", \\"t\\".\\"id\\" as \\"id\\" from \\"txs\\" as \\"t\\" where \\"sender\\" = 'sender' and \\"t\\".\\"time_stamp\\" >= 'timeStart' and \\"t\\".\\"time_stamp\\" <= 'timeEnd' order by \\"t\\".\\"time_stamp\\" asc, \\"t\\".\\"id\\" asc limit 1"`;

exports[`Sql builder search covers case with all filters with after 1`] = `"select \\"t\\".\\"tx_type\\" as \\"tx_type\\", \\"t\\".\\"id\\" as \\"id\\" from \\"txs\\" as \\"t\\" where \\"sender\\" = 'sender' and (t.time_stamp, t.id) > ('timestamp', 'id') and \\"t\\".\\"time_stamp\\" >= 'timeStart' and \\"t\\".\\"time_stamp\\" <= 'timeEnd' order by \\"t\\".\\"time_stamp\\" asc, \\"t\\".\\"id\\" asc limit 1"`;
